app/models/mixins/ownership_mixin.rb

```diff
@@ -2,13 +2,16 @@ module OwnershipMixin
   extend ActiveSupport::Concern
 
   included do
-    belongs_to :evm_owner, :class_name => "User"
+    belongs_to :evm_owner, :class_name => "User", :foreign_key => :evm_owner_id
     belongs_to :miq_group
 
-    virtual_column :evm_owner_email,                      :type => :string,     :uses => :evm_owner
-    virtual_column :evm_owner_name,                       :type => :string,     :uses => :evm_owner
-    virtual_column :evm_owner_userid,                     :type => :string,     :uses => :evm_owner
-    virtual_column :owned_by_current_user,                :type => :boolean,    :uses => :evm_owner_userid
+    virtual_delegate :email, :name, :userid, :to => :evm_owner, :prefix => true, :allow_nil => true
+    virtual_attribute :owned_by_current_user, :boolean, :uses => :evm_owner, :arel => (lambda do |t|
+      #t.grouping(arel_attribute(:evm_owner_id)]).eq(User.current_user.try(:id)))
+      t.grouping(
+        Arel::Nodes::NamedFunction.new("LOWER", [arel_attribute(:evm_owner_userid)]).eq(User.current_userid.downcase)
+      )
+    end)
```

Comment:
Well, I know this isn't the sexy answer, but seeing all the issues with having a `virtual_delegate` being called from another `virtual_attribute`, would doing the following work for now?

```ruby
virtual_attribute :owned_by_current_user, :boolean, :arel => (lambda do |t|
  user_table = User.arel_table
  group_sel  = t.grouping(user_table[:userid].lower.eq(User.current_userid.downcase))
  where_cond = user_table[:id].eq(arel_attribute(:evm_owner_id))

   t.grouping(user_table.project(group_sel).where(where_cond))
end)
```